lib/src/chart/line_chart/line_chart_data.dart

```diff
@@ -1653,6 +1661,9 @@ class LineBarSpot extends FlSpot with EquatableMixin {
   /// Is the index of our [super.spot], in the [LineChartBarData.spots] list.
   final int spotIndex;
 
+  /// Distance in pixels from where the user taped
+  final double distance;
```

Comment:
Why do we need this?
`LineBarSpot` is a general class. It is not only for touched spots. It's a wrapper to hold a spot location.

Reply:
Hmm, yes you are right. My intention was to give more control to the user, who then can choose which spots to select. But maybe for now I should just go with your suggestion, to return the LineBarSpots as they are returned currently, along with the LineBarSpot that is the closest. (As described here: https://github.com/imaNNeoFighT/fl_chart/pull/891#issuecomment-1029808580)

Let me take a look at that.